src/tikzdocument.jl

```diff
@@ -142,17 +142,15 @@ _HAS_WARNED_SHELL_ESCAPE = false
 function savepdf(filename::String, td::TikzDocument;
                  latex_engine = latexengine(),
                  buildflags = vcat(DEFAULT_FLAGS, CUSTOM_FLAGS),
-                 run_count = 0)
+                 run_count = 0, tmp = tempname())
     global _HAS_WARNED_SHELL_ESCAPE, _OLD_LUALATEX
     run_again = false
 
-    tmp = tempname()
     tmp_tex = tmp * ".tex"
     tmp_pdf = tmp * ".pdf"
-    savetex(tmp_tex, td)
+    if run_count == 0 savetex(tmp_tex, td) end
```

Comment:
Perhaps write this as `run_count == 0 && savetex(tmp_tex, td)`, I think we do that in the rest of the code.